pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -780,6 +808,12 @@ def _handle_final_result(
 
         # For backwards compatibility, append a new ModelRequest using the tool returns and retries
         if tool_responses:
+            # Only apply templates if explicitly configured
+            if ctx.deps.prompt_config and ctx.deps.prompt_config.templates is not None:
```

Comment:
As above, I prefer to write these things as `if (x := y.z) and (foo := x.bar)` etc, so that we can just use `templates` below instead of repeating the method chain